contracts/common/misc/DelegateProxy.sol

```diff
@@ -0,0 +1,44 @@
+pragma solidity ^0.5.5;
+
+import { Ownable } from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+import { ProxyData } from './ProxyData.sol';
+import { ERCProxy } from './ERCProxy.sol';
+
+contract DelegateProxy is ProxyData, Ownable, ERCProxy {
+  constructor(address _proxyTo)
+    public
+    Ownable()
+  {
+    proxyTo = _proxyTo;
+  }
+
+  function proxyType() external pure returns (uint256 proxyTypeId) {
+    // Upgradeable proxy
+    proxyTypeId = 2;
+  }
+
+  function implementation() external view returns (address codeAddr) {
+    codeAddr = proxyTo;
+  }
+
+  function updateProxyTo(address _proxyTo)
+    public
+    onlyOwner
+  {
+    proxyTo = _proxyTo;
```

Comment:
Simple log `event ProxyUpdated(address indexed _new, address indexed _old, address indexed _owner)` (added `_owner` for receipt in case owner changes) would be great